flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/exec/BatchExecNode.scala

```diff
@@ -32,4 +33,13 @@ trait BatchExecNode[T] extends ExecNode[BatchPlanner, T] with Logging {
     */
   def getDamBehavior: DamBehavior
 
+  def setManagedMemoryWeight[X](
+      transformation: Transformation[X], memoryBytes: Long): Transformation[X] = {
+    // Using Bytes can easily overflow
+    // Using MebiBytes to cast to int
+    // Careful about zero
+    val memoryMB = Math.max(1, (memoryBytes >> 20).toInt)
```

Comment:
Seems the weight cannot be set to zero with this max agg.

Reply:
An alternative is to make the min weight to 2 if `memoryBytes` > 0. 
In this way we do not need to set the weight for operators that do not use managed memory, by overriding value 1 weights to 0 in `ExecutorUtils#setBatchProperties`, like what we do for ResourceSpec.
It's a bit hacky though. But cannot be easily replaced once job wide default ResourceSpec is supported.

Reply:
Yeah, It's a bit hacky, looks like we create these hacky code to work around runtime default value.
Because our current `ManagedMemoryWeight` settings are very large. So I think default 1 is very small and is not a serious problem, so I prefer just check in `ExecutorUtils#setBatchProperties`, and warn log when weight is default 1.

Reply:
Actually I'd prefer the way mentioned in another comment (https://github.com/apache/flink/pull/10452#discussion_r354659007) which is not hacky. It would be good to have a unified entrypoint to create transformations so that table layer can easily apply defaults it wants.

Reply:
I will try my best to set memory to all operators, but the memory of source/sink may not guarantee to set zero. By default, 1 can't steal much memory. It's OK to leave some manage memory for source/sink.

Reply:
source & sink will be the bad case of managed memory, do we have a way to also set the managed memory to 0 for source & sink?

Reply:
Hi @zhuzhurk , we can not control all transformations, because some of them come from user's Source/Sink.

Hi @KurtYoung 
- 1. Current Transformation not have an unknown value to indicate that the weight not be set.
The thing is that we can not control the default value of Transformation. If runtime let the default value of Transformation's weight is `null`, in this way, we can set all null to zero, we can control.
- 2. The another way is just as @zhuzhurk said, we can consider default 1 to be unknown value, in this way, we set all 1 to 0. But that is a little hacky.
- 3. The third way is just let it go. By default, 1 can't steal much memory (we set sort/join/agg to 128). We can even set KibiBytes, then the sort/join/agg is 128 * 1024.

I prefer option 3.

Reply:
Just to confirm whether users' custom sources/sinks would use managed memory?
If user source/sink would not use any managed memory, table can also override user operators' managed memory size/weight to 0. (table can see all operators in the StreamGraph and knows which operators are from table, the others should be source/sink operators I think)
If user the source/sink is possible to use managed memory, then table should guarantee that those user operators would still be able to acquire manage memory, considering the competition from managed memory weights of table operators. Otherwise those source/sink tasks may fail.